Warn when the token query param is used for auth (#16009)
hashicorp/consul: website/content/docs/upgrading/upgrade-specific.mdx

The `connect.enable_serverless_plugin` configuration option was removed. Lambda integration is now enabled by default.

#### Deprecating authentication via token query parameter

> When authenticating API requests, the use of the `token` query parameter is deprecated. The Consul token should be provided to API requests using the `X-Consul-Token` header or with the
> Bearer scheme in the authorization header.
> 
> Before Consul v1.15: 
> 
> ```shell-session
> $ curl \
>     http://127.0.0.1:8500/v1/agent/members?token=<consul token>
> ```
> 
> In Consul v1.15 and higher:
> 
> ```shell-session
> $ curl \
>     --header "X-Consul-Token: <consul token>" \
>     http://127.0.0.1:8500/v1/agent/members
> ```

Reviewer: This may imply that passing via the `token` query parameter was expected before Consul v1.15, which isn't accurate.

The `X-Consul-Token` header has been supported for a while and has been the recommended way. Now we're explicitly deprecating the token query param.

We should also mention the log message that indicates if the token query param is in use.

Perhaps change to something like:

````suggestion
Providing a Consul ACL token in API requests using the `token` query parameter is deprecated and will be removed in Consul 1.17.
Instead, the token should be provided using the `X-Consul-Token` header or with the Bearer scheme in the authorization header as described in the [API authentication documentation](/consul/api-docs/api-structure#authentication).

To check whether the `token` query parameter is in use, search your Consul agent logs for the message:
```shell-session
<INSERT WHAT THE MESSAGE IS>
```

Deprecated authentication using the `token` query parameter:

```shell-session
$ curl \
    http://127.0.0.1:8500/v1/agent/members?token=<consul token>
```

Recommended usage:

```shell-session
$ curl \
    --header "X-Consul-Token: <consul token>" \
    http://127.0.0.1:8500/v1/agent/members
```
````

The `connect.enable_serverless_plugin` configuration option was removed. Lambda integration is now enabled by default.

#### Deprecating authentication via token query parameter

Providing a Consul ACL token in API requests using the `token` query parameter is deprecated and will be removed in Consul 1.17.
Instead, the token should be provided using the `X-Consul-Token` header or with the Bearer scheme in the authorization header as described in the [API authentication documentation](/consul/api-docs/api-structure#authentication).
	
To check whether the `token` query parameter is in use, search your Consul agent logs for the message:

```shell-session

> $ This request is using a token query parameter which is deprecated and will be removed in Consul 1.17

Reviewer: ```suggestion
$ This request is using the token query parameter which is deprecated and will be removed in Consul 1.17
```

Reviewer 2: nit: The suggestion in http.go says "This request **used** the token..." I think "used" is the word that we want here, not "uses".. Just pointing this out for consistency.

> Instead, the token should be provided using the `X-Consul-Token` header or with the Bearer scheme in the authorization header as described in the [API authentication documentation](/consul/api-docs/api-structure#authentication).
> 	
> To check whether the `token` query parameter is in use, search your Consul agent logs for the message:
> 
> ```shell-session
> $ This request is using a token query parameter which is deprecated and will be removed in Consul 1.17
> ```

Reviewer 3: ````suggestion
Instead, you should provide the token through the `X-Consul-Token` header or with the Bearer scheme in the authorization header as described in the [API authentication documentation](/consul/api-docs/api-structure#authentication).
	
Check whether you are using the `token` query parameter by searching your Consul agent logs for the message:

```shell-session hideClipboard
$ This request is using a token query parameter which is deprecated and will be removed in Consul 1.17
````
```

Reviewer: "the token query param" rather than "a token query param"
